types/sharedb/lib/client.d.ts

```diff
@@ -9,6 +9,18 @@ export class Connection {
     // ShareDB, but it is handy for server-side only user code that may cache
     // state on the agent and read it in middleware
     agent: Agent | null;
+
+    collections: Record<string, Record<string, Doc>>;
+    queries: Record<string, Query>;
+
+    seq: number;
+    id: string;
```

Comment:
Should be `string | null`, and I'd also add JSDoc with the same comment in the source, `/** Equals agent.src on the server */`.

Reply:
(I personally find `| null` to be noise in TS, because every property can be set to `null`, so it doesn't really add anything, but fine)